tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/SendMessageBatchAsyncIntegration.cs

```diff
@@ -64,7 +66,15 @@ public class SendMessageBatchAsyncIntegration
                 for (int i = 0; i < requestProxy.Entries.Count; i++)
                 {
                     var entry = requestProxy.Entries[i].DuckCast<IContainsMessageAttributes>();
-                    ContextPropagation.InjectHeadersIntoMessage<TSendMessageBatchRequest>(entry, scope.Span.Context);
+
+                    var dataStreamsManager = Tracer.Instance.TracerManager.DataStreamsManager;
+                    if (dataStreamsManager != null && dataStreamsManager.IsEnabled)
```

Comment:
`queueName` could be null. Should we skip this if so?

```suggestion
                    if (!string.IsNullOrEmpty(queueName)
                      && Tracer.Instance.TracerManager.DataStreamsManager is { IsEnabled: true } dataStreamsManager)
```